src/components/Chat/InputMessageV2.tsx

```diff
@@ -286,18 +291,23 @@ const InputMessageV2 = ({
     if (success) {
       ImagePicker.openCamera({
         mediaType: 'photo'
-      }).then(async (imageRes) => {
-        const imageCropped = await ImagePicker.openCropper({
-          mediaType: 'photo',
-          path: imageRes.path,
-          width: imageRes.width,
-          height: imageRes.height,
-          cropperChooseText: 'Next',
-          freeStyleCropEnabled: true
+      })
+        .then(async (imageRes) => {
+          setIsLoadingUploadImageCamera(false);
+          const imageCropped = await ImagePicker.openCropper({
+            mediaType: 'photo',
+            path: imageRes.path,
+            width: imageRes.width,
+            height: imageRes.height,
+            cropperChooseText: 'Next',
+            freeStyleCropEnabled: true
+          });
+          refAttachment.current.close();
+          handleUploadCamera(imageCropped.path);
+        })
+        .catch(() => {
+          setIsLoadingUploadImageCamera(false);
         });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, the permission check before opening the camera is missing in the new code. It's important to ensure that the app has the necessary permissions before attempting to access the camera.

```diff
+    const {success} = await requestCameraPermission();
+    if (success) {
         ImagePicker.openCamera({
           mediaType: 'photo'
         })
         .then(async (imageRes) => {
           setIsLoadingUploadImageCamera(false);
           // rest of the code...
         })
         .catch(() => {
           setIsLoadingUploadImageCamera(false);
         });
+    } else {
+      setIsLoadingUploadImageCamera(false);
+      openAlertPermission(
+        'We’re not able to access your camera, please adjust your permission settings for BetterSocial.'
+      );
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->